test/stateful_dataloader/test_state_dict.py

```diff
@@ -841,5 +847,49 @@ def test_two_dataloaders(self) -> None:
         self.assertEqual(data, exp)
 
 
+class TestJsonSerDe(unittest.TestCase):
+    def _run_test(self, num_workers):
+        interrupt = 4
+        dataset = DummyIterableDataset([0, 100, 37], shuffle=False, include_generator=False)
```

Comment:
Do we need test case for map style with worker state too?

Reply:
For our existing dummy tests, this wouldn't make a difference since non-RNG state only exists for the sampler, which shouldn't fall in to this category. I could add it if you feel strongly

Reply:
@andrewkho Hmm maybe I misunderstood something here - why does the state being RNG vs non-RNG matter for this change? For example, DummyMapDataset which has state (though RNG one) should be considered as worker state isn't it?

Reply:
@gokulavasan it's for trivial reasons: rng state is not json-serializable so we can't include it in this test :( but you know what, I just had another look and realized that the sampler also does have json-serializable state, no RNG, so I'll add that test too